src/lib/optimizer/strategy/column_pruning_rule.cpp

```diff
@@ -85,9 +85,23 @@ ExpressionUnorderedSet gather_locally_required_expressions(
     case LQPNodeType::Aggregate: {
       const auto& aggregate_node = static_cast<AggregateNode&>(*node);
 
-      // Tracks whether we have already added a column. For COUNT(*) without GROUP BY, we might otherwise end up
-      // without any expressions.
-      auto has_at_least_one_expression = false;
+      // Handling COUNT(*) (which is represented as an LQPColumnReference with a valid original_node and an
+      // INVALID_COLUMN_ID) is difficult, as we need to make sure that at least one expression from that
```

Comment:
Why does at least one expression need to survive?

Reply:
Further the note sounds like "we stop once we see star" while the code is "if we see count star, take all expressions".

Reply:
> Handling COUNT(*) (which is represented as an LQPColumnReference with a valid original_node and an INVALID_COLUMN_ID) is difficult, as we need to make sure that at least one expression from that original node survives the pruning. Otherwise, we could not resolve that original_node later on. For now, we simply stop pruning (i.e., add all expressions as required) once we encounter COUNT(*).

better?